[aoti] Change aot_compile callsites

angelayi opened the pull request:
Summary:
Replacing `torch._export.aot_compile` callsites with 
```
ep = torch.export._trace._export(.., predispatch=True)   # Traces the given program into predispatch IR
so_path = torch._inductor.aot_compile_ep(ep, ...)  # Takes an exported program and compiles it into a .so
```

This allows us to explicitly split up the export step from AOTInductor. We can later modify tests to do `export + serialize + deserialize + inductor` to mimic internal production use cases better.

Test Plan: CI

Differential Revision: D54808612




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

desertfire reviewed (commented):
CI failures look real.

khabinov commented on torch/_inductor/__init__.py:
```diff
@@ -90,6 +90,70 @@ def aot_compile(
     )
 
 
+def aot_compile_ep(
```
Thanks for giving this name! Helps us avoid confusions.

angelayi committed:
[aoti][1/n] Change aot_compile callsites (#122225)

Summary:
X-link: https://github.com/pytorch/benchmark/pull/2207


Replacing `torch._export.aot_compile` callsites with 
```
ep = torch.export._export(.., predispatch=True)   # Traces the given program into predispatch IR
so_path = torch._inductor.aot_compile_ep(ep, ...)  # Takes an exported program and compiles it into a .so
```

This allows us to explicitly split up the export step from AOTInductor. We can later modify tests to do `export + serialize + deserialize + inductor` to mimic internal production use cases better.

This PR also enables export's predispatch IR most of the the existing use cases. Previously this is using export to torch IR, which is a different graph. This may result in some performance regressions as some of inductor's passes will no longer run -- if so, please let me know.

This PR changes the seemingly low-impact files to the new calling convention, and a followup PR will change the high-impact site.

Test Plan: CI

Reviewed By: SherlockNoMad

Differential Revision: D54808612
